org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/AbstractDeclarativeIdeQuickfixProvider.java

```diff
@@ -49,7 +58,20 @@ private boolean getFixMethodPredicate(Method input, String issueCode) {
 		return false;
 	}
 
-	private List<DiagnosticResolution> getResolutions(Options options, Iterable<Method> fixMethods) {
+	private DiagnosticResolution getResolution(Options options, Iterable<Method> fixMethods, int id) {
+		DiagnosticResolutionAcceptor issueResolutionAcceptor = issueResolutionAcceptorProvider.get();
+		List<Method> methods = Lists.newArrayList(fixMethods);
```

Comment:
This is based on the assumption that the methods will be returned in the correct order always.
Given that the order in Class.getMethods is not defined, we might see flaky tests where the expectations contains the int id.

Reply:
you're right, it would be better to tag the methods with ids as suggested. i'll consider this.